src/aqua/chain/new_deals.aqua

```diff
@@ -0,0 +1,143 @@
+module NewDeals declares poll_new_deals
+
+import or, not from "@fluencelabs/aqua-lib/binary.aqua"
+import Spell, TriggerConfig from "@fluencelabs/spell/spell_service.aqua"
+
+import FluenceAuroraConnector, DealCreated from "services.aqua"
+
+import AuroraInfo from "aurora.aqua"
+import deal_log, spell_log, get_counter, get_string from "../fluence/spell.aqua"
+import is_worker_created from "../fluence/worker.aqua"
+import Json from "../fluence/peer.aqua"
+import join_deal from "../decider/join_deal.aqua"
+import SpellId from "../types.aqua"
+import need_update_from_block from "../chain/blocks.aqua"
+
+-- Get latest block number from RPC
+func get_latest(spell_id: SpellId, chain: AuroraInfo) -> ?string:
+    latest: ?string
+    r <- FluenceAuroraConnector.latest_block_number(chain.api_endpoint)
+    if r.success:
+        latest <<- r.result
+    else:
+        spell_log(spell_id, ["error retrieving latest block number", r.error])
+    <- latest
+
+
+-- `last seen block` is a block for which all events have been processed
+func get_last_seen(spell_id: SpellId) -> ?string:
+    Spell spell_id
+    last_seen, ok <- get_string(spell_id, "last_seen_block")
```

Comment:
It doesn't look like _refactoring_ but more like _redoing_. Maybe don't do it here, plz? It's hard to review.

Reply:
we can have a call and go through everything together, no problem :)

After some thinking, I came to conclusion that `from_block` is a mix of several things:
1. "last seen" block number, ie a block of which decider has processed some events
2. `init_data.from_block`, a configuration parameter
3. left boundary of block range that decider wants to scan

So I tried to name them separately, and it was inevitable for such things
```
        if result.result.length > 1 == false:
            need_update <- need_update_from_block(info.api_endpoint, result.to_block)
            if need_update:
                log(["updating outdated from_block: [previous from_block, new_from_block]", from_block, result.to_block])
                to_block_str <- Json.stringify(result.to_block)
                Spell.set_string("from_block", to_block_str)
```

to be rewritten